HPXMLtoOpenStudio/resources/EPvalidator.xml

```diff
@@ -0,0 +1,1131 @@
+      <sch:assert test='count(h:extension/h:FracLatent) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/FracLatent</sch:assert>
+      <sch:assert test='count(h:extension/h:UsageMultiplier) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/UsageMultiplier</sch:assert>
+      <sch:assert test='count(h:extension/h:WeekdayScheduleFractions) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/WeekdayScheduleFractions</sch:assert>
+      <sch:assert test='count(h:extension/h:WeekendScheduleFractions) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/WeekendScheduleFractions</sch:assert>
+      <sch:assert test='count(h:extension/h:MonthlyScheduleMultipliers) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/MonthlyScheduleMultipliers</sch:assert>
+    </sch:rule>
+
+    <!-- [FuelLoad] -->
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:MiscLoads/h:FuelLoad[h:FuelLoadType="grill" or h:FuelLoadType="lighting" or h:FuelLoadType="fireplace"]'>
+      <sch:assert test='count(h:SystemIdentifier) = 1'>Expected 1 element(s) for xpath: SystemIdentifier</sch:assert> <!-- Required by HPXML schema -->
+      <sch:assert test='count(h:Location[text()="interior" or text()="exterior"]) &lt;= 1'>Expected 0 or 1 element(s) for xpath: Location[text()="interior" or text()="exterior"]</sch:assert>
+      <sch:assert test='count(h:Load[h:Units="therm/year"]/h:Value) &lt;= 1'>Expected 0 or 1 element(s) for xpath: Load[Units="therm/year"]/Value</sch:assert>
+      <sch:assert test='count(h:FuelType[text()="natural gas" or text()="fuel oil" or text()="fuel oil 1" or text()="fuel oil 2" or text()="fuel oil 4" or text()="fuel oil 5/6" or text()="diesel" or text()="propane" or text()="kerosene" or text()="coal" or text()="coke" or text()="bituminous coal" or text()="anthracite coal" or text()="wood" or text()="wood pellets"]) = 1'>Expected 1 element(s) for xpath: FuelType[text()="natural gas" or text()="fuel oil" or text()="fuel oil 1" or text()="fuel oil 2" or text()="fuel oil 4" or text()="fuel oil 5/6" or text()="diesel" or text()="propane" or text()="kerosene" or text()="coal" or text()="coke" or text()="bituminous coal" or text()="anthracite coal" or text()="wood" or text()="wood pellets"]</sch:assert>
+      <sch:assert test='count(h:extension/h:UsageMultiplier) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/UsageMultiplier</sch:assert>
+      <sch:assert test='count(h:extension/h:WeekdayScheduleFractions) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/WeekdayScheduleFractions</sch:assert>
+      <sch:assert test='count(h:extension/h:WeekendScheduleFractions) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/WeekendScheduleFractions</sch:assert>
+      <sch:assert test='count(h:extension/h:MonthlyScheduleMultipliers) &lt;= 1'>Expected 0 or 1 element(s) for xpath: extension/MonthlyScheduleMultipliers</sch:assert>
+    </sch:rule>
+  </sch:pattern>
+  
+  <sch:pattern name='[Neighbors]'>
```

Comment:
@bpark1327 You have some `sch:rule`s pulled out into separate `sch:pattern`s as needed. I'm sure it works correctly, but I think most developers will find it difficult to understand/know when to extract into a separate `sch:pattern` and when not to. Is there any reason not to have _every_ `sch:rule` in its own `sch:pattern`? That would both simplify the file and as well as allow these `sch:pattern`s you've pulled out to be reinserted at a more logical location (e.g., WindowOverhang adjacent to Window) rather than simply tacked to the end of the file.

Reply:
Good point. There's actually no reason not to have every `sch:rule` in its own `sch:pattern`. As far as I know, we can create as many patterns as we like. I agree that having every `sch:rule` in its own `sch:pattern` would have a lot of benefits, as you mentioned. I'll address it.